src/site/content/en/blog/constructable-stylesheets/index.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: post
+title: Constructable Stylesheets
+subhead: Seamless reusable styles.
+authors:
+  - developit
+date: 2019-02-08
+updated: 2022-02-23
+description: Constructable Stylesheets provide a seamless way to create and distribute styles to documents or shadow roots without worrying about FOUC.
+tags:
+  - blog
+  - css
+---
+
+[Constructable Stylesheets](https://wicg.github.io/construct-stylesheets) are a
+way to create and distribute reusable styles when using [Shadow
+DOM](https://developers.google.com/web/fundamentals/web-components/shadowdom).
+
+{% Aside %}
+Constructable Stylesheets are available in Chromium
+(versions 73 and higher), and in Firefox from version 75 behind the
+`layout.css.constructable-stylesheets.enabled` flag.
+{% endAside %}
+
+It has always been possible to create stylesheets using JavaScript. However, the
+process has historically been to create a `<style>` element using
+`document.createElement('style')`, and then access its sheet property to obtain
+a reference to the underlying
+[CSSStyleSheet](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet)
+instance. This method can produce duplicate CSS code and its attendant bloat,
+and the act of attaching leads to a flash of unstyled content whether there is
+bloat or not. The `CSSStyleSheet` interface is the root of a collection of CSS
+representation interfaces referred to as the
+[CSSOM](https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model),
+offering a programmatic way to manipulate stylesheets as well as eliminating the
+problems associated with the old method.
+
+{% Img
+src="image/ZDZVuXt6QqfXtxkpXcPGfnygYjd2/ZGONaG0tsbRHWzCZpAlo.png",
+alt="Diagram showing preparation and application of CSS.",
+width="800", height="610" %}
+
+Constructable Stylesheets make it possible to define and prepare shared CSS
+styles, and then apply those styles to multiple Shadow Roots or the Document
+easily and without duplication. Updates to a shared CSSStyleSheet are applied to
+all roots into which it has been adopted, and [adopting a
+stylesheet](https://wicg.github.io/construct-stylesheets/#using-constructed-stylesheets)
+is fast and synchronous once the sheet has been loaded.
+
+The association set up by Constructable Stylesheets lends itself well to a
+number of different applications. It can be used to provide a centralized theme
+used by many components: the theme can be a `CSSStyleSheet` instance passed to
+components, with updates to the theme propagating out to components
+automatically. It can be used to distribute [CSS Custom
+Property](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) values to
+specific DOM subtrees without relying on the
+[cascade](https://developer.mozilla.org/en-US/docs/Web/CSS/Cascade). It can even
```

Comment:
```suggestion
Property](https://developer.mozilla.org/docs/Web/CSS/--*) values to
specific DOM subtrees without relying on the
[cascade](https://developer.mozilla.org/docs/Web/CSS/Cascade). It can even
```